core-web/apps/dotcms-ui/src/app/api/services/dot-announcements.ts/dot-announcements.service.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable, Signal, computed, inject } from '@angular/core';
+import { toSignal } from '@angular/core/rxjs-interop';
+
+import { catchError, pluck, tap } from 'rxjs/operators';
+
+export type Announcement = {
+    title: string;
+    type: string;
+    announcementDateAsISO8601: string;
+    identifier: string;
+    url: string;
+};
+
+export enum TypesIcons {
+    Comment = 'pi pi-comment',
+    Release = 'pi pi-book',
+    Announcement = 'pi pi-megaphone'
+}
+
+@Injectable()
+export class AnnouncementsService {
+    private http = inject(HttpClient);
+    private announcementsUrl = '/api/v1/announcements';
+
+    private announcementsData$ = this.http.get<Announcement[]>(this.announcementsUrl).pipe(
+        pluck('entity'),
+        tap((announcements: Announcement[]) => {
+            announcements.map((announcement) => {
+                announcement.url = `${announcement.url}?utm_source=dotcms&utm_medium=application&utm_campaign=announcement_menu`;
+            });
+
+            return announcements;
+        }),
+
+        catchError(() => [])
+    );
+
+    announcements: Signal<Announcement[]> = toSignal(this.announcementsData$, {
+        initialValue: [] as Announcement[]
+    });
+
+    unreadAnnouncements: Signal<boolean> = computed(() => {
+        const storedAnnouncementsJson = localStorage.getItem('announcementsData');
+        const storedAnnouncements: Announcement[] = storedAnnouncementsJson
+            ? JSON.parse(storedAnnouncementsJson)
+            : [];
+
+        const newAnnouncements = this.announcements();
+        const newAnnouncementIds = newAnnouncements.map((announcement) => announcement.identifier);
+        const storedAnnouncementIds = storedAnnouncements.map(
+            (announcement) => announcement.identifier
+        );
+
+        const isNewAnnouncement = newAnnouncementIds.some(
+            (id) => !storedAnnouncementIds.includes(id)
+        );
+
+        return isNewAnnouncement;
+    });
+
+    saveAnnouncementsData: Signal<void> = computed(() => {
```

Comment:
I think a computed needs to return a value (You can no return, but the purpose of the computed is compute a value based on other value), here is used as effect
Maybe we can add an effect on service constructor, or take other strategy to update the localStorage

Reply:
Indeed if a computed signal is a void, it should be an effect.

Reply:
TIL!